### PR TITLE
Nexus Task Definitions

### PR DESCRIPTION
**Pull request by bergundy**
## What changed?

Added all of the task definitions and additions to mutable state for Nexus callbacks.

## Why?

Prep work for Nexus callbacks.

## How did you test it?

Added relevant unit tests.

### CODE REVIEW

File: service/history/shard/context_impl.go [hunk 1]
```diff
@@ -1684,6 +1684,10 @@ func (s *ContextImpl) notifyQueueProcessor() {
 	now := s.timeSource.Now()
 	fakeTasks := make(map[tasks.Category][]tasks.Task)
 	for _, category := range s.taskCategoryRegistry.GetCategories() {
+		// TODO: remove this as soon as we've implemented the callback queue processor.
+		if category == tasks.CategoryCallback {
+			continue
+		}
```
Comment (yycptt) [inline service/history/shard/context_impl.go:1690]: nit: probably just don't register the new category before the queue processor is ready.

File: service/history/tasks/utils.go [hunk 2]
```diff
@@ -120,6 +120,8 @@ func GetTimerTaskEventID(
 		eventID = common.FirstEventID
 	case *DeleteHistoryEventTask:
 		eventID = common.FirstEventID
+	case *CallbackBackoffTask:
+		eventID = common.FirstEventID
```
[the remark this answers is not in the document]
Comment (bergundy) [reply service/history/tasks/utils.go:124]: I didn't seem to need this in my PoC.
I will add it if required.

File: service/history/tasks/callback_backoff_task.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	"go.temporal.io/server/common/definition"
+)
+
+// CallbackBackoffTask is a timer task for backing off after failed callback attempts.
+type CallbackBackoffTask struct {
+	definition.WorkflowKey
+	VisibilityTimestamp time.Time
+	TaskID              int64
+	EventID             int64
```
[the remark this answers is not in the document]
Comment (bergundy) [reply service/history/tasks/callback_backoff_task.go:37]: Hmm... yeah, I don't need this AFAICT.

File: service/history/shard/context_impl.go (hunk 1 above)
Comment (bergundy) [reply service/history/shard/context_impl.go:1690]: Good point.

File: service/history/tasks/utils.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (yycptt) [reply service/history/tasks/utils.go:124]: oh right. it's not a transfer task... 🤦

**Review by yycptt — APPROVED** [review]

**Pull request merged**